### ISSUE TITLE
Bug: Missing Images

### ISSUE THREAD

**Issue by LeonardoGentile** [HappenApps/Quiver#204]
I'm not sure if this happened because of dropbox or because of the beta version but I've just realized I'm missing some images within some notes. Others are fine and appear correctly
Any hint to debug or restore the missing images?

![quiver](https://cloud.githubusercontent.com/assets/412061/10612868/ff3a6834-7752-11e5-8aaa-747770ad3b2a.png)

Thanks

**Comment by ylian** [comment]
Can you send me the note file?

Was the content (including images) copied from a webpage? If so, let me know the link. If not, were they copied from another app?

I fixed a couple issues recently related to this. It's likely that this is already fixed.

**Comment by ylian** [comment]
It's also possible that the note was created using an early beta. I used a different image url format in one of the early betas. In that case it should be very easy to fix.

**Comment by ylian** [comment]
I think most likely you created the note using an early beta that used a different image url format.

Check the note's "content.json" file. If you see images like this:

``` html
<img src=\"resources/9B7E1A72-B213-41DB-B4BC-567F8B76199E.jpg\">
```

Change it to:

``` html
<img src=\"quiver-image-url/9B7E1A72-B213-41DB-B4BC-567F8B76199E.jpg\">
```

You can open your Quiver library in a text editor and do a global search and replace.

**Comment by LeonardoGentile** [comment]
Hello,
I've checked and the images urls are fine, using the second format 

```
<img src=\"quiver-image-url/9B7E1A72-B213-41DB-B4BC-567F8B76199E.jpg\">
```

I can see 4 occurences of images with this format within `content.json`, the problem is that 2 of them are contained inside the `resources` directory (I can see them in the note) and the other 2 are missing (can't see them from the note).

I'm syncing with dropbox, so I've tried to look if and when these were deleted but there isn't any trace of them, at least for the last 30 days (last time I've edited that note was more than 30 days ago) so I can't really say what happened to them.

**Comment by LeonardoGentile** [comment]
@ylian you still need the file to take a look?

**Comment by ylian** [comment]
That's weird. Please send over the note.

**Comment by ylian** [comment]
Thanks for the note. The note content does look fine, but somehow those two images are missing from the resources folder.

The only thing I can think of is a bug I fixed in beta 9: if two notes are open in two windows (e.g., one in the main window, the other in a separate window), copying images from one to another won't copy over the image file.

If that's not what happened, please keep an eye on the issue. There might still be an edge case I didn't handle.

**Comment by LeonardoGentile** [comment]
@ylian thanks for your help. 
I'm not sure if the case you described was also my case because I've updated the note some time ago and I can't remember, but it might be possible, indeed. 

I will take a look to similar note to see if I can find the original images.

Btw, regarding the case you described, do the original images have the same name of the destination ones? I'm aware that they are stored separate note packages but also their name is totally different?
In this case I could look by file name

**Comment by ylian** [comment]
If you copy an image from one note to another, the image name will be the same. I have changed the image name to MD5 instead of UUID in the latest beta. So identical images will have the same names from now on.

**Comment by LeonardoGentile** [comment]
@ylian I've found weird things.
On my second machine I opened the faulty note and the images were there!
I actually opened the note from finder and the images were **physically** there:
![fa979b14-8b4f-4b65-8b0e-e76a66704e1a_qvnote](https://cloud.githubusercontent.com/assets/412061/10695899/749f2e20-79a6-11e5-9516-21b50ba23908.png)

Then I opened a terminal and looked for the missing (?) images with: 

```
find . -name "0BD41AF9-6852-****-****-20D4834467EC.png"
# found the same images in two different notes

find . -name "8CDEF85F-****-****-97E4-000BE0B226E1.png"
# as before found the same images in in two different notes, always the same.
```

My guess is that it is due to the bug you've mentioned but also I guess a sort of bug in dropbox because otherwise how could it be that the images are physically present in one machine and not in another?

Tonight I'll check the faulty note on the other machine to see what's going on there

btw should I be superparanoid offuscating UUID or not? :grimacing:

**Comment by ylian** [comment]
You are not seeing ghosts. I implemented a self-recover mechanism for images in beta 9, so when you open the note on your second machine, Quiver went to look for the missing images from other notes and found them, then copied them over.

It seems that the missing images were caused by the bug I mentioned. Clearly the images were copied from another note. I will mark this as closed.

And you don't need to obfuscate UUID. But it's totally okay to be paranoid about security these days.
